### PR TITLE
gh-520: Removed unnecassary types from S1 

### LINKED ISSUE #520: CLEAN:  Remove some unnecessary types from `S1`
I think that `S1` has types that are not needed, e.g., the `datetime` types.  

As discussed here https://github.com/pandas-dev/pandas-stubs/pull/519#pullrequestreview-1271164311

we should see which ones we can safely remove without changing any tests or other parts 

For things like `Timestamp`, `Timedelta`, `Period`, we may be able to get rid of all references to `Series[Timestamp]`, `Series[Timedelta]` and `Series[Period]` because we have `TimestampSeries`, `TimedeltaSeries` and `PeriodSeries`, but we leave the references just in the class definition there from an inheritance perspective.

@ramvikrams to investigate

### PR DESCRIPTION
**Pull request by ramvikrams**
- [x] Closes #520 
- [ ] Tests added: Please use `assert_type()` to assert the type of any return value

I didn't delete `datetime.date , dateitme.time` it was called many times in funcs

### CODE REVIEW

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Based on the table you did, we should add `np.timedelta64`  to `S1`

In `pandas-stubs/_libs/tslibs/timestamps.pyi`, for `__eq__()` and `__ne__()` replace `Series[Timestamp]` with `Series[np.datetime64]`.  Then in `test_scalars.py`, change line 1261 to use `Series[np.datetime64]` instead of `Series[pd.Timestamp]`.  I checked this and it will work.

In `pandas-stubs/_libs/tslibs/timestamps.pyi`, you have a `Series[Timedelta]` for `__add__()` that you should remove.

Otherwise, this looks good.

File: pandas-stubs/_typing.pyi [hunk 1]
```diff
@@ -197,9 +197,7 @@ S1 = TypeVar(
     str,
     bytes,
     datetime.date,
-    datetime.datetime,
     datetime.time,
-    datetime.timedelta,
```
[the remark this answers is not in the document]
Comment (ramvikrams) [reply pandas-stubs/_typing.pyi:202]: Actually to remove both of them we would have to do some changes in ` core\indexes\accessors.pyi` and change a test in `test_timefuncs.py`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
So it appears that we should not remove `dt.date` and `dt.time` now that I see the tests you had to change.
```python
>>> import pandas as pd
>>> i0 = pd.date_range(start="2022-06-01", periods=10)
>>> i0
DatetimeIndex(['2022-06-01', '2022-06-02', '2022-06-03', '2022-06-04',
               '2022-06-05', '2022-06-06', '2022-06-07', '2022-06-08',
               '2022-06-09', '2022-06-10'],
              dtype='datetime64[ns]', freq='D')
>>> s0 = pd.Series(i0)
>>> s0
0   2022-06-01
1   2022-06-02
2   2022-06-03
3   2022-06-04
4   2022-06-05
5   2022-06-06
6   2022-06-07
7   2022-06-08
8   2022-06-09
9   2022-06-10
dtype: datetime64[ns]
>>> s0.dt.date
0    2022-06-01
1    2022-06-02
2    2022-06-03
3    2022-06-04
4    2022-06-05
5    2022-06-06
6    2022-06-07
7    2022-06-08
8    2022-06-09
9    2022-06-10
dtype: object
>>> s0.dt.date.iloc[0]
datetime.date(2022, 6, 1)
>>> s0.dt.time
0    00:00:00
1    00:00:00
2    00:00:00
3    00:00:00
4    00:00:00
5    00:00:00
6    00:00:00
7    00:00:00
8    00:00:00
9    00:00:00
dtype: object
>>> s0.dt.time.iloc[0]
datetime.time(0, 0)
>>> s0.dt.timetz
0    00:00:00
1    00:00:00
2    00:00:00
3    00:00:00
4    00:00:00
5    00:00:00
6    00:00:00
7    00:00:00
8    00:00:00
9    00:00:00
dtype: object
>>> s0.dt.timetz.iloc[0]
datetime.time(0, 0)
```

So the  tests here:
```python
    check(assert_type(s0.dt.date, "pd.Series[dt.date]"), pd.Series, dt.date)
    check(assert_type(s0.dt.time, "pd.Series[dt.time]"), pd.Series, dt.time)
    check(assert_type(s0.dt.timetz, "pd.Series[dt.time]"), pd.Series, dt.time)
``` 
were correct, because what is inside the `Series` are `dt.date` and `dt.time` objects.

So can you revert the changes you made corresponding to `dt.date` and `dt.time` ?

Sorry for misleading you.

**Comment by ramvikrams** [comment]
> 
> So can you revert the changes you made corresponding to `dt.date` and `dt.time` ?
 
Yes sir

> 
> Sorry for misleading you.

No problem Sir, everyone of us makes mistake All thanks to you for having this great attitude

**Review by Dr-Irv — APPROVED** [review]
Thanks @ramvikrams

**Comment by ramvikrams** [comment]
> Thanks @ramvikrams

thanks sir

**Pull request merged**